sherpa/astro/xspec/__init__.py

```diff
@@ -723,24 +758,71 @@ def set_xsxsect(name: str) -> None:
     _xspec.set_xsxsect(name)
 
 
+def clear_xsxset() -> None:
+    """Remove all existing X-Spec model settings.
+
+    .. versionadded:: 4.17.1
+
+    See Also
+    --------
+    get_xsxset, set_xsxset
+
+    Examples
+    --------
+
+    >>> set_xsxset("POW_EMIN", "0.5")
+    >>> get_xsxset()
+    {'POW_EMIN': '0.5'}
+    >>> clear_xsxset()
+    >>> get_xsxset()
+    {}
+
+    """
+    return _xspec.clear_xsxset()
+
+
+@overload
+def get_xsxset() -> dict[str, str]:
+    ...
+
+@overload
+def get_xsxset(name: None) -> dict[str, str]:
+    ...
+
+@overload
 def get_xsxset(name: str) -> str:
-    """Return the X-Spec model setting.
+    ...
+
+def get_xsxset(name: str | None = None) -> str | dict[str, str]:
+    """Return the X-Spec model setting or settings.
+
+    .. versionchanged:: 4.17.1
+       This routine can now be called with no argument, which means
+       that a dictionary containing all the settings is returned.
+       Asking for an unset keyword now raises a KeyError rather than
+       returning the empty string.
 
     Parameters
     ----------
-    name : str
+    name : str or None, optional
        The name of the setting (converted to upper case before being
        sent to X-Spec). There is no check that the name is valid.
 
     Returns
     -------
-    val : str
-       Returns the value set by a previous call to `set_xsxset` or the
-       empty string, if the value has not been previously set.
+    val : str or dict
+       When name is set, returns the value set by a previous call to
+       `set_xsxset`. When name is not set a dictionary of all the
+       settings is returned.
+
+    Raises
+    ------
+    KeyError
+       When the name has not been set.
```

Comment:
```suggestion
       When the name passed is not set in X-Spec.
```
(See the text just a few lines above "When the name is not set a dictionary ... is returned", so we can't also have that it raises and error is not set.)